api/utils/requestProcessor.js

```diff
@@ -2582,11 +2594,13 @@ const ignorePossibleDevices = (params) => {
 /**
  * Restart Request
  * @param {params} params - params object
+ * @param {function} initiator - function which initiated request
  * @param {function} done - callback when processing done
  * @param {number} try_times - how many times request was retried
+ * @param {function} fail - callback when restart limit reached
  * @returns {void} void
  */
-const restartRequest = (params, done, try_times) => {
+const restartRequest = (params, initiator, done, try_times, fail) => {
```

Comment:
Codacy found an issue: ['restartRequest' was used before it was defined.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6152164)